include/revng/Model/CommonTypeMethods.h

```diff
@@ -0,0 +1,224 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ADT/RecursiveCoroutine.h"
+#include "revng/Model/PrimitiveKind.h"
+
+namespace model {
+class VerifyHelper;
+
+class TypeDefinition;
+class TypedefDefinition;
+class EnumDefinition;
+class StructDefinition;
+class UnionDefinition;
+
+class Type;
+class ArrayType;
+class DefinedType;
+class PointerType;
+class PrimitiveType;
+
+template<typename CRTP>
+class CommonTypeMethods {
+public:
+  /// Recursively computes size of the type.
+  ///
+  /// It asserts in cases where the size cannot be computed, for example, when
+  /// the type system loops and the type's size depends on the type itself.
+  ///
+  /// \returns * `std::nullopt` if the type does not have size (for example,
+  ///            it's a `void` primitive or a function type),
+  ///          * size in bytes otherwise.
+  std::optional<uint64_t> size() const debug_function;
+  RecursiveCoroutine<std::optional<uint64_t>> size(VerifyHelper &VH) const;
+
+  /// Tries to recursively compute the size of the type.
+  ///
+  /// Use this method only on (temporarily) invalid models. You most likely want
+  /// to use size().
+  ///
+  /// \returns * `std::nullopt` if the size cannot be computed, for example,
+  ///            when the type system loops and the type's size depends on
+  ///            the type itself,
+  ///          * 0 for types without the size (`void` and function types),
+  ///          * size in bytes in all other cases.
+  std::optional<uint64_t> trySize() const debug_function;
+  RecursiveCoroutine<std::optional<uint64_t>> trySize(VerifyHelper &VH) const;
+
+public:
+  /// This is a strict version of \ref skipToDefinition. This does not ignore
```

Comment:
Please do not refer to `skipToDefinition`. Just state what the function, defining it by difference is very confusing.

Reply:
Done!